модель проекта запроса баланса

koterinka opened the pull request:
+ README.md
+ pom.xml
+ gitignore
+ project model

koterinka committed:
+ gitignore
+ README.md
+ pom.xml
+ модель проекта запроса баланса в jdk

RodinDenis commented on src/main/java/ru/atm/ATM.java:
```diff
@@ -0,0 +1,14 @@
+
+public class ATM {
+
+    public BigDecimal GetCardBalance(CardInfo card, int pinCode)
```
я бы рекомендовал заложить в дизайн сущность баланс, которая могла бы объединить сумму баланса и валюту

RodinDenis commented on src/main/java/ru/atm/CardInfo.java:
```diff
@@ -0,0 +1,14 @@
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class CardInfo {
```
я бы рекомендовал заложить в дизайн этого класса также атрибут пин

RodinDenis commented on src/main/java/ru/atm/Human.java:
```diff
@@ -0,0 +1,20 @@
+        System.out.println("My balance is: " + atm.GetCardBalance(this.card, cardPin));
+    }
+
+    private CardInfo card;
```
поля и конструктор я рекомендую привыкать писать в шапке класса

koterinka committed:
+lomboc
+slf4j
+interface
private fields
system.out.print change logger.info

RodinDenis commented on src/main/java/ru/atm/ATM.java:
```diff
@@ -0,0 +1,11 @@
+
+public class ATM implements Application {
+    @Override
+    public BigDecimal GetCardBalance(CardInfo card, String pinCode) {
```
имя метода лучше начинать с буквы в нижнем регистре. getCardBalance

RodinDenis commented on src/main/java/ru/atm/Application.java:
```diff
@@ -0,0 +1,7 @@
+import java.math.BigDecimal;
+
+public interface Application {
+    BigDecimal GetCardBalance(CardInfo Card, String CardPin);
```
параметры и имя метода тоже лучше писать в нижнем регистре card, cardPin

RodinDenis commented on src/main/java/ru/atm/Main.java:
```diff
@@ -0,0 +1,22 @@
+        CardInfo cardInfo= new CardInfo("12345678",ExpirationDate);
+        Human human = new Human(cardInfo, "4321");
+        ATM atm= new ATM();
+        Logger logger = LoggerFactory.getLogger(Main.class);
```
обычно выносят в поля  класса, а лучше спрятать за аннотацией

koterinka committed:
геттерв, сеттеры над классом
имя и параметры метода в нижнем регистре
logger public static под шапкой класса
application app (вместо atm)
